### DISCUSSION TITLE
Issue on update from 0.1.27

### DISCUSSION THREAD

**Discussion by westberliner** [vitest-dev/vitest#1142] [Q&A] [answered]
Hi,

I have a problem to upgrade from 0.1.27 and I'm not sure how to solve this. Any ideas would be very helpful.
My setup is Vue3 with Vuex, vitest, jsdom, vue-testutils. If I upgrade I'll get on some tests errors, especially on integration tests with child components. It does not matter if I'll try 0.2.0 or 0.9.3. The result is the same. It differs if I'll switch to happy-dom but not by much.

While debugging I noticed that `wrapper.find('#some-element').trigger('some-event')` does not work but if I try it with `findComponent('#some-element').trigger` it works. Another issue is in some tests `findComponent('#some-element')` return null . But with find by module `findComponent(MyModule)` it works. `setValue` won't work as well. but if I'll use `dom.element.value = 'someValue'` I'll get around. But not my goal if I want to test input events as well.
It feels like jsdom and the vue components are disconnected and the test cannot interact with the mounted app or component.

Any ideas on where to look at? Or what could course this between 0.1.27 and 0.2.0?

Thx
Patrick

**Comment by sheremet-va** [comment, marked as answer]
Notable difference between 0.1.x and 0.2.x is changing behaviour of  fake timers. Do you use `vi.useFakeTimers()`?

**Comment by westberliner** [comment]
Yes, I do. But I am not sure how this could course these issues. On top of that some tests with `vi.useFakeTimers()` are still working.
Thx for the quick reply.

**Reply by sheremet-va** [reply]
We also don't know why yet. 
See https://github.com/vitest-dev/vitest/issues/649

**Reply by westberliner** [reply]
Wow, crazy. Thanks for pointing that out. I use fake timers only for a debounce method. I'll try to mock the function and come back to confirm this issue.

**Reply by westberliner** [reply]
Tested. In all cases, this solves my issues by avoiding fake timers. thx for pointing that out. never thought this would cause such behavior.
